features/steps/oauth.py

```diff
@@ -118,6 +119,37 @@ def step_impl(context):
         assert False, error
 
 
+@when('I authorize the app {action}ing access to {resource_type}')
+def step_impl(context, action, resource_type):
+
+    assert action in AUTHORIZATION_ACTIONS
+    assert resource_type in s4s.MU_CCDS_MAPPINGS
+
+    # Filter the steps to only what is required for this authorization
+    condition = '{0}.{1}'.format(action, resource_type)
+    steps = context.vendor_config['auth'].get('steps', [])
+    steps = [step for step in steps
+             if 'tag' not in step or step['tag'] == condition]
```

Comment:
I went with `tag` as the conditional name, but it doesn't quite feel right because it's not really behaving the way tags behave. I'd like to use something more indicative of the toggling behavior, (maybe `when`?), but I figured I'd run it by you before I made Allscripts rewrite their PR because I'm particular about how I name things.